pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -187,7 +187,6 @@ func (p *Provider) loadConfigurationFromIngresses(ctx context.Context, client Cl
 	serverVersion, err := client.GetServerVersion()
 	if err != nil {
 		log.FromContext(ctx).Errorf("Failed to get server version: %v", err)
-		return conf
```

Comment:
won't this ignoring of error here could result in improper processing of IngressClass resource semantics. Assume an installation with IngressClass resource defined and an Ingress resource referring to that.  Under scenarios where kube-api is unavailable, this would cause the `shouldProcessIngress()` method to skip such an ingress in the provider sync loop.

Reply:
@ravilr that is a great scenario that I hadn't considered.

Reply:
I have taken your suggestion, and reworked this to kick the error back to the provider loop.